testing/mock.ts

```diff
@@ -1413,11 +1413,6 @@ export async function assertSpyCallAsync<
           "do not expect error and return, only one should be expected",
         );
       }
-      if (call.error) {
-        throw new AssertionError(
-          "spy call did not return expected value, an error was thrown.",
-        );
-      }
```

Comment:
When `call.error` is truthy, `assertSpyCallAsync` throws earlier before reaching here.